### PR TITLE
Initial Upgrade to CakePHP 5.x

### PR DESCRIPTION
**Pull request by arusinowski**
Hi, this is fixturize plugin upgraded to CakePHP 5.0, to merge it the `3.next` branch need to be created. Please review. I will update this branch when the CakePHP5 is released. Thanks

### CODE REVIEW

**Comment by ADmad** [comment]
Please revert the unnecessary changes to `.editorconfig` and `.gitignore`.

File: composer.json [hunk 1]
```diff
@@ -34,11 +34,14 @@
         "cs-fix": "phpcbf --colors --standard=vendor/cakephp/cakephp-codesniffer/CakePHP src/"
     },
     "require": {
-        "cakephp/cakephp": "^4.0"
+        "php": ">=8.1",
```
Comment (ravage84) [inline composer.json:47]: This is not necessary. Since we fully depend on the CakePHP, the framework defines the lower and upper PHP bounds.

**Comment by ravage84** [comment]
The static analysis error `Call to an undefined method Cake\Datasource\ConnectionInterface::execute().` should be addressed

**Review by ravage84 — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (arusinowski) [reply composer.json:47]: Change applied, thanks

File: src/TestSuite/Fixture/ChecksumTestFixture.php [hunk 2]
```diff
@@ -21,27 +23,27 @@ class ChecksumTestFixture extends TestFixture
      *
      * @var array<string, string>
      */
-    protected static $_tableHashes = [];
+    protected static array $tableHashes = [];
```
Comment (ADmad) [inline src/TestSuite/Fixture/ChecksumTestFixture.php]: The removal of underscore from the properties and methods is unnecessary noise.

Comment (arusinowski) [reply src/TestSuite/Fixture/ChecksumTestFixture.php]: restored

File: composer.json [hunk 3]
```diff
@@ -30,14 +30,25 @@
...
+        "cs-fix": "phpcbf --colors --parallel=16 -p src/",
+        "phpstan": "tools/phpstan analyse",
+        "psalm": "tools/psalm --show-info=false",
+        "stan": [
+            "@phpstan",
+            "@psalm"
+        ],
+        "stan-baseline": "tools/phpstan --generate-baseline",
+        "psalm-baseline": "tools/psalm  --set-baseline=psalm-baseline.xml",
+        "stan-setup": "phive install",
```
Comment (ravage84) [inline composer.json]: This is potentially more than just the setup of PHPStan.

I'd call it `phive-install`, `phive-setup` or more generally `tools-setup`, instead.

File: src/TestSuite/Fixture/ChecksumTestFixture.php (hunk 2 above)
Comment (ravage84) [reply src/TestSuite/Fixture/ChecksumTestFixture.php]: @arusinowski the following methods are still changed:

- `_getTableKey()`
- `_tableUnmodified()`
- `_hash()`

**Review by ravage84 — CHANGES_REQUESTED** [review]

File: src/TestSuite/Fixture/ChecksumTestFixture.php (hunk 2 above)
Comment (arusinowski) [reply src/TestSuite/Fixture/ChecksumTestFixture.php]: restored

File: composer.json (hunk 3 above)
Comment (arusinowski) [reply composer.json]: done

**Review by ravage84 — APPROVED** [review]

**Pull request merged**